spec/omniauth-dsds/lib/user_builder_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "spec_helper"
+require_relative "../../../lib/omniauth-dsds/lib/user"
+require_relative "../../../lib/omniauth-dsds/lib/user_builder"
+require "omniauth"
+
+RSpec.describe Omniauth::Dsds::UserBuilder do
+  subject { Omniauth::Dsds::UserBuilder.new app_id: app_id, app_secret: app_secret }
+  let(:app_id) { "12345" }
+  let(:app_secret) { "ABCDE" }
+  let(:token) { "ZXCVB" }
+
+  describe "#build_user" do
+    context "with a blank token" do
+      let(:token) { nil }
+
+      it "returns nil" do
+        expect(subject.build_user(token)).to be_nil
+      end
+    end
+
+    before do
+      allow(subject).to receive(:fetch_raw_info).with(token).and_return raw_info_response
```

Comment:
Yeah design. Trying not to touch private methods, and just want to test the 2 code paths when the roles array has values, and when it does not...